sound/soc/sof/intel/hda-dai.c

```diff
@@ -158,6 +155,7 @@ static int hda_link_hw_free(struct snd_pcm_substream *substream,
 	if (!link)
 		return -EINVAL;
 
+	snd_soc_dai_set_dma_data(dai, substream, NULL);
```

Comment:
what exactly does this do here? What information are you trying to save?

Reply:
Here is just to clear saving. The dma data is save at hda_link_hw_params, and clear at hda_link_hw_free

Reply:
does this below in the same patch? what is the link with the DMA link configuration, it looks like a miss in previous patches rather?

Reply:
yes, it is missed by previous patches. I just check this data and found it should be cleared.